commit 5ac76caa6945edf0f07197a848fce4ae31ebdf30 (refs/changes/22/5622/2)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2019-02-22T13:58:38-08:00 (8 months ago)
    
    TRITON-1242 Create DAPI filter to allow filtering out virtual servers
    Reviewed by: Marsell Kukuljevic <marsell@joyent.com>
    Approved by: Marsell Kukuljevic <marsell@joyent.com>

diff --git a/docs/index.md b/docs/index.md
index 210f13b..76725d0 100644
--- a/docs/index.md
+++ b/docs/index.md
@@ -123,6 +123,7 @@ specialized circumstances in production.
 | **ALLOC_WEIGHT_UNRESERVED_DISK**   | Float | 1.0   | Bias selection towards CNs with more unreserved disk.                        |
 | **ALLOC_WEIGHT_UNRESERVED_RAM**    | Float | 2.0   | Bias selection towards CNs with more unreserved memory.                      |
 | **FEATURE_USE_CNAGENT_COMMAND_EXECUTE** | Boolean | false | Experimental: Use cn-agent's command_execute function instead of Ur when available. |
+| **FEATURE_ENABLE_VIRTUAL_SERVERS_FILTER** | Boolean | false | Whether to enable the virtual server filtering. When enabled, during DAPI allocation, all virtual servers (e.g. mockcloud servers) will be filtered out if the vm is a docker vm, or the vm has the `triton.placement.exclude_virtual_servers` tag set to true. This virtual server filtering is needed for Triton testing, but should not be enabled for production environments. |
 
 If any of the keys above aren't in the `sdc` `metadata` section, it's treated as
 if the default value was specified. Be careful when changing from the default
diff --git a/package.json b/package.json
index 285b84d..07afbbe 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cnapi",
   "description": "SmartDataCenter Compute Node API",
-  "version": "1.19.4",
+  "version": "1.20.0",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
@@ -9,7 +9,7 @@
     "assert-plus": "1.0.0",
     "async": "1.5.2",
     "bunyan": "1.8.5",
-    "dapi": "git+https://github.com/joyent/sdc-designation.git#e70e8e91f97496a7df3140dd061b1a48b73d3251",
+    "dapi": "git+https://github.com/joyent/sdc-designation.git#35914310ca9b359ad64b95ab4ea42b763399694a",
     "deep-equal": "1.0.1",
     "dox": "0.9.0",
     "gc-stats": "1.2.0",
diff --git a/sapi_manifests/cnapi/template b/sapi_manifests/cnapi/template
index 69a2643..2dc3f6b 100644
--- a/sapi_manifests/cnapi/template
+++ b/sapi_manifests/cnapi/template
@@ -79,6 +79,9 @@
 			        "hard-filter-running",
 			        "hard-filter-invalid-servers",
 			        "hard-filter-force-failure",
+{{#FEATURE_ENABLE_VIRTUAL_SERVERS_FILTER}}
+			        "hard-filter-virtual-servers",
+{{/FEATURE_ENABLE_VIRTUAL_SERVERS_FILTER}}
 			        "hard-filter-volumes-from",
 			        "hard-filter-reserved",
 			        "hard-filter-min-free-disk",
